csrc/scheduler/transpose.cpp

```diff
@@ -587,6 +628,67 @@ std::string getTransposeRuntimeRejectReason(
   }
 #endif
 
+  // TODO: ideally we shouldn't have to manually match schedule transformation
+  // here. It is hard to maintain consistent code logic.
+  if (!scheduler_utils::getViewTVs(fusion).empty()) {
+    const auto index_type = runtime_info.getIndexType();
+    auto params =
+        std::make_shared<TransposeParams>("Transpose heuristics", index_type);
+    maybeBuildVirtualInnerDims(
+        *params,
+        device_multiprocessor_count,
+        n_elems,
+        shape_in_ref1,
+        inner_most_pos1_in_ref1,
+        inner_most_pos2_in_ref1);
+
+    // disallow transpose scheduler when we have a combination of:
+    // 1. view op; and
+    // 2. small transpose transformation
+    // See note [Supporting small transpose dimensions]
+    if (hasSmallTransposeDimensions(params)) {
+      return "Small transpose dimensions and view op cannot be currently be handled by transpose scheduler. See: https://github.com/NVIDIA/Fuser/pull/592";
+    }
+
+    // mimic transform propagation
```

Comment:
Don't review the whole PR yet. But does this part looks about right?

Since we are not applying cacheBefore and cacheAfter yet, so this is likely not the real path that was propagated later during scheduling. :cry: 
Would this be good enough?

Reply:
cc'ing @zasdfgbnm

Reply:
I think it should be good enough.

Reply:
I backed out the last two traversal check, since those are only propagating the tiling logic, which shouldn't really introduce any incoherent transformation. And more importantly those are rejecting nanogpt cases...